apps/edit/app.lisp

```diff
@@ -76,32 +71,42 @@
 (defun-bind mouse-cursor (mouse_xy)
 	(defq buffer (elem text_buffer current_text))
 	(bind '(ox oy cx cy sx) (elem text_position current_text))
-
 	(defq cursor_xy (list cx cy) char_wh (vdu-char-size vdu) offset_xy (list ox oy))
 	(setq cursor_xy (map + (map / mouse_xy char_wh) offset_xy)
 		cx (elem 0 cursor_xy) cy (elem 1 cursor_xy))
-	(if (>= cy (length buffer)) (setq cy (min cy (dec (length buffer))) cx (length (elem cy buffer))))
+	(if (>= cy (length buffer)) 
+		(setq cy (min cy (dec (length buffer))) cx (length (elem cy buffer))))
 	(if (> cx (length (elem cy buffer)))
 		(setq cx (max (set-sticky) (length (elem cy buffer)))) (setq sx cx))
-
 	(elem-set text_buffer current_text buffer)
 	(elem-set text_position current_text (list ox oy cx cy sx)))
 
 (defun-bind open-buffer (path)
-	(defq index (length text_store) title path buffer (list (join " " (ascii-char 10))) pos (list 0 0 0 0 0))
-	(unless (or (eql path "") (not (catch (file-stream path) t)))
```

Comment:
(ascii-char 10)

(const (ascii-char 10)) for faster code as const will evaluate during read time. Did you notice BTW my comments on the "Textfields and Editor" issue for you ?

Chris

Reply:
I did. I tried to add the num-to-utf8 function and the text ended up blank. I'm not feeling great today. I've got just about everything cleaned up as you suggested. I went through and added const to all the structure references as well.

Reply:
Hope you feel better soon !

Did you set the 'font property of the buttons, or toolbar flow they are in, to the Entypo font ? Maybe that's why they showed up blank, only the Entypo font has those symbol chars in it and the default font for buttons is the OpenSans one.